lib/templates/deploy.batch.go.template

```diff
@@ -0,0 +1,68 @@
+goto Deployment
+
+:: Utility Functions
+:: -----------------
+
+:SelectGoVersion
+
+IF DEFINED KUDU_SELECT_GO_VERSION_CMD (
+    :: TODO
+goto :EOF
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+:: Deployment
+:: ----------
+
+:Deployment
+ECHO Handling Go deployment.
+
+:: 1. Prepare workspace
+IF "%GOROOT%" == "" (
+    :: select go version TODO
+    SET GOROOT=D:\Program Files\go\1.4.2
+)
+ECHO GOROOT %GOROOT%
+
+SET GOPATH=%DEPLOYMENT_TEMP%\gopath
+SET GOEXE="%GOROOT%\bin\go.exe"
+SET FOLDERNAME=azureapp
+SET GOAZUREAPP=%DEPLOYMENT_TEMP%\gopath\src\%FOLDERNAME%
+
+IF EXIST %GOPATH% (
+    ECHO %GOPATH% already exist
+    
+    ECHO Removing %GOAZUREAPP%
+    RMDIR /S /Q %GOAZUREAPP%
+) else (
+    ECHO Creating %GOPATH%\bin
+    MKDIR "%GOPATH%\bin"
+    
+    ECHO Creating %GOPATH%\pkg
+    MKDIR "%GOPATH%\pkg"
+    
+    ECHO Creating %GOPATH%\src
+    MKDIR "%GOPATH%\src"
+)
+
+ECHO Creating %GOAZUREAPP%
+MKDIR %GOAZUREAPP%
+
+:: 2. KuduSync
+ECHO Prepare source files for build
+IF /I "%IN_PLACE_DEPLOYMENT%" NEQ "1" (
```

Comment:
when %IN_PLACE_DEPLOYMENT% is set we should probably fail as it is not supported for go.